src/coreclr/jit/emit.cpp

```diff
@@ -1006,6 +1006,15 @@ insGroup* emitter::emitSavIG(bool emitAdd)
         assert(emitLastIns != nullptr);
         assert(emitCurIGfreeBase <= (BYTE*)emitLastIns);
         assert((BYTE*)emitLastIns < emitCurIGfreeBase + sz);
+
+#if defined(TARGET_XARCH)
+        assert(emitLastIns != nullptr);
+        if (emitLastIns->idIns() == INS_jmp && (((instrDescJmp*)emitLastIns)->idjIsRemovableJmpCandidate))
+        {
+            ig->igFlags |= IGF_HAS_REMOVABLE_JMP;
```

Comment:
You can remove the `idjIsRemovableJmpCandidate` check here. We will be checking that anyway in `emitRemoveJumpToNextInst()`.

Reply:
Done.